interp/interp.go

```diff
@@ -361,30 +366,64 @@ func (interp *Interpreter) resizeFrame() {
 	interp.frame.data = data
 }
 
-func (interp *Interpreter) main() *node {
-	interp.mutex.RLock()
-	defer interp.mutex.RUnlock()
-	if m, ok := interp.scopes[mainID]; ok && m.sym[mainID] != nil {
-		return m.sym[mainID].node
-	}
-	return nil
-}
-
 // Eval evaluates Go code represented as a string. Eval returns the last result
 // computed by the interpreter, and a non nil error in case of failure.
 func (interp *Interpreter) Eval(src string) (res reflect.Value, err error) {
 	return interp.eval(src, "", true)
 }
 
-// EvalPath evaluates Go code located at path. EvalPath returns the last result
-// computed by the interpreter, and a non nil error in case of failure.
-func (interp *Interpreter) EvalPath(path string) (res reflect.Value, err error) {
-	// TODO(marc): implement eval of a directory, package and tests.
-	b, err := ioutil.ReadFile(path)
-	if err != nil {
-		return res, err
+// EvalPath evaluates Go code located at path, running tests specified by tfe
+// which is the test filter regular expression. If tfe is empty, the main function
+// is evaluated. EvalPath returns the last result computed by the interpreter,
+// and a non nil error in case of failure.
+func (interp *Interpreter) EvalPath(path string, skiptest bool) (res reflect.Value, err error) {
```

Comment:
I'm not super happy about the extra boolean everywhere obviously, but there does not seem to be any better alternative. :( 
Maybe for once we could consider a variadic...

Reply:
The alternate is to keep `EvalPath(path)` as is and add `EvalTestPath(path)`.

Reply:
yes, much better imho.